ui/src/kapacitor/containers/TickscriptPage.js

```diff
@@ -172,55 +173,77 @@ class TickscriptPage extends Component {
       } else {
         response = await createTask(kapacitor, task, router, sourceID)
       }
-
-      if (response && response.code === 500) {
-        return this.setState({validation: response.message})
+      if (response && !response.code) {
```

Comment:
In what case will `response` be missing here? This would read way better if you could enforce its presence

```
if (response.code) {
  // unsavedChanges: true
} else {
  // unsavedChanges false
}
```

Also is `response.code` a boolean??

Reply:
Ah that's a good point -- there's always a response when you try to save. You only get `response.code` if there's an error so the presence of that is how we've been differentiating between a failed or successful save. I think your suggestion is cleaner